source/GameInterface/Services/ItemRosters/ItemRosterRegistry.cs

```diff
@@ -0,0 +1,24 @@
+﻿using GameInterface.Services.Registry;
+using System.Threading;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.Roster;
+using TaleWorlds.CampaignSystem.Settlements;
+
+namespace GameInterface.Services.ItemRosters
+{
+    internal class ItemRosterRegistry : RegistryBase<ItemRoster>
+    {
+        private const string ItemRosterIdPrefix = "CoopItemRoster";
+        private static int ItemRosterCounter = 0;
+        public ItemRosterRegistry(IRegistryCollection collection) : base(collection) { }
+
+        public override void RegisterAll()
+        {
+        }
+
+        protected override string GetNewId(ItemRoster itemRoster)
+        {
+            return $"{ItemRosterIdPrefix}_{Interlocked.Increment(ref ItemRosterCounter)}";
```

Comment:
Please use GUID here, also need to register all itemrosters on game loaded

Reply:
JK dont do use GUIDs this, we need these to be the same on the server and clients on startup, but still need register all